controllers/datadogagent/merger/fake/volume_mount_manager.go

```diff
@@ -23,12 +25,25 @@ func (_m *VolumeMountManager) AddVolumeMount(volumeMount *v1.VolumeMount) {
 
 // AddVolumeMountToContainer provides a mock function with given fields: volumeMount, containerName
 func (_m *VolumeMountManager) AddVolumeMountToContainer(volumeMount *v1.VolumeMount, containerName commonv1.AgentContainerName) {
-	_m.VolumeMountsByC[containerName] = append(_m.VolumeMountsByC[containerName], volumeMount)
+	isInitContainer := false
+	for _, initContainerName := range initContainerNames {
+		if containerName == initContainerName {
+			isInitContainer = true
+			break
+		}
+	}
+	if !isInitContainer {
+		_m.VolumeMountsByC[containerName] = append(_m.VolumeMountsByC[containerName], volumeMount)
+	}
 }
 
 // AddVolumeMountToInitContainer provides a mock function with given fields: volumeMount, containerName
 func (_m *VolumeMountManager) AddVolumeMountToInitContainer(volumeMount *v1.VolumeMount, containerName commonv1.AgentContainerName) {
-	_m.VolumeMountsByC[containerName] = append(_m.VolumeMountsByC[containerName], volumeMount)
+	for _, initContainerName := range initContainerNames {
+		if containerName == initContainerName {
+			_m.VolumeMountsByC[containerName] = append(_m.VolumeMountsByC[containerName], volumeMount)
+		}
+	}
```

Comment:
Same as the above mocked envvar manager https://github.com/DataDog/datadog-operator/pull/893/files#r1302201156, but with the mocked volumeMount manager. Actual volumeMount manager [behavior](https://github.com/DataDog/datadog-operator/blob/480f69931540a7de1808ffd7bd9b5043c1249fe7/controllers/datadogagent/merger/volume_mount.go#L43-L57).